### PR TITLE
Custom and registry cell type documentation

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/2250

### CODE REVIEW

**Review by lzdanski — APPROVED** [review]
Woo! A straight up reference doc!

I took a look at the preview and this content, and I can go back through with a super fine toothed editing comb, but I only really saw some passive voice that wasn't interfering with effective communication. 

The Only concerns I have (which are non-blocking IMO). Tables are difficult for screenreaders and with time can get really unruly in the source code, sometimes columns or bulleted lists work better, but it's a tradeoff. We'll have to think about it and maybe explore a bit!

File: astro/cloud-ide/use-airflow-operators.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+sidebar_label: Use Airflow operators
+title: Use Airflow operators in the Astro Cloud IDE
+id: use-airflow-operators
+description: Learn how to run Airflow operators by creating and configuring operator cells in the Astro Cloud IDE.
+---
+
+You can import any Airflow operator available on the Astronomer Registry and run it as part of your Cloud IDE pipeline. In most cases, the Cloud IDE can parse parameters and provides you with a detailed form for configuring the operator. 
+
+## Create an operator cell
```
Comment (jlaneve) [inline astro/cloud-ide/use-airflow-operators.md:10]:
```suggestion
## Using open-source Airflow operators
```

File: astro/cloud-ide/use-airflow-operators.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+![A boolean formatted as a literal value](/img/cloud-ide/literal-value.png)
+
+If you click the toggle on and then select **Python expression**, you see that the IDE converts the toggle into a raw Python expression.
+
+![A boolean formatted as a Python expression](/img/cloud-ide/python-expression.png)
+
+:::
+
+## Create custom operator cells
```
Comment (jlaneve) [inline astro/cloud-ide/use-airflow-operators.md:44]:
```suggestion
## Using custom Airflow operators
```

File: astro/cloud-ide/use-airflow-operators.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+To import a custom operator that your team uses into the Astro Cloud IDE, create a custom cell type in your project configuration.
+
+1. If you're using a custom cell type to call custom operator code, [commit your IDE project to a Git repository](deploy-project.md#commit-your-project-to-a-git-repository) and add your custom operator code to the `include` folder of your repository.
```
Comment (jlaneve) [inline astro/cloud-ide/use-airflow-operators.md:48]:
```suggestion
1. If you're using a custom cell type to call custom operator code, [link your IDE project to a Git repository](deploy-project.md#commit-your-project-to-a-git-repository) and add your custom operator code to the `include` folder of your repository.
```

File: astro/cloud-ide/custom-cell-reference.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+## Display JSON reference
+
+The following table is a reference for all possible key-value pairs in the `display` section of the JSON request.
+
+
+| Name          | Type   | Description                                       |
+| ------------- | ------ | ------------------------------------------------- |
+| `label`       | String | The label to show in the **Add Cell** menu.       |
+| `description` | String | The description to show in the **Add Cell** menu. |
```
Comment (jlaneve) [inline astro/cloud-ide/custom-cell-reference.md:18]:
```suggestion
| Name          | Type   | Description                                       |
| ------------- | ------ | ------------------------------------------------- |
| `label`       | String | The label to show in the **Add Cell** menu.       |
| `description` | String | The description to show in the **Add Cell** menu. |
| `logoUrl` | String | The URL of the logo to show in the **Add Cell** menu. |
```

File: astro/cloud-ide/custom-cell-reference.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+| `invoke`    | Key-value pairs                              | The configuration for how the operator is invoked. Set `functionName` (String), `imports` (List of Strings), and `excludeTaskArgId` (Boolean)                                                                                                                         |
+| `namingStrategy`   | String, either `increment` or `uuid` | Determines how the default names are generated for new cells.                                                                                                                      |
+| `excludeFromGraph` | Boolean                              | Determines whether to hide instances of the cell from the graph view of your pipeline editor.                                                                                      |
+| `returnsRawValue`  | Boolean                              | Indicates whether the custom operator returns a raw value. Set to `true` if the operator uses TaskFlow. If set to false, `.output` is appended to the name of the operator output. |
```
Comment (jlaneve) [inline astro/cloud-ide/custom-cell-reference.md:44]:
```suggestion
| Name               | Type                                 | Description                                                                                                                                                                        |
| ------------------ | ------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `type`         | String, only option is `invoke`.                              | Determines how the operator is called.                                                                                                                                      |
| `invoke`    | Key-value pairs                              | The configuration for how the operator is invoked. Set `functionName` (String), `imports` (List of Strings), and `excludeTaskArgId` (Boolean)                                                                                                                         |
```

File: astro/cloud-ide/custom-cell-reference.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+| Name               | Type                                 | Description                                                                                                                                                                        |
+| ------------------ | ------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `label`         | String                             | The label for the parameter.                                                                                                                                      |
+| `description`    | String                              | The description for the parameter.                                                            |
+| `default`   | Key-value pair. The key becomes the default `dataType` and the value becomes the default value. | The default type and value for the parameter.                                                                                                                      |
```
Comment (jlaneve) [inline astro/cloud-ide/custom-cell-reference.md:70]: we don't actually support this yet
```suggestion
```

File: astro/cloud-ide/custom-cell-reference.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+| `example` | Key-value pair. The key becomes the default `dataType` and the value becomes the default value.                              | The example value that appears in placeholder text.                                                                                      |
+| `fixedWidthFont`  | Boolean                              | Determines whether to format input as a fixed-width font. |
```
Comment (jlaneve) [inline astro/cloud-ide/custom-cell-reference.md:72]: we don't actually support this yet
```suggestion
```

File: astro/cloud-ide/custom-cell-reference.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+## Configs JSON reference
+
+The following table is a reference for all possible key-value pairs in the `configs` section of the JSON request.
+
+You can configure multiple `configs` objects for each parameter you want the user to enter. 
+
+| Name               | Type                                 | Description                                                                                                                                                                        |
+| ------------------ | ------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `key`         | String                             | The unique identifier for the parameter.                                                                                                                                      |
+| `dataType`    | String, either `string`, `stringList`, `boolean`, `integer`                              | The parameter's value type.                                                              |
```
Comment (jlaneve) [inline astro/cloud-ide/custom-cell-reference.md:56]:
```suggestion
| `dataType`    | String, must be one of: `string`, `stringList`, `stringMap`, `stringListMap`, `stringMapList`, `stringSet`, `boolean`, `integer`, `float`, `duration`, `unsupported`                              | The parameter's value type.                                                              |
```

**Review by jlaneve — APPROVED** [review]
some minor tweaks - will probably want to continue adding to and improving this over the next week or so, but this is a great first draft!

File: astro/cloud-ide/use-airflow-operators.md (hunk 1 above)
Comment (jwitz) [inline astro/cloud-ide/use-airflow-operators.md:10]:
```suggestion
## Create an Airflow operator cell
```

Comment (jwitz) [reply astro/cloud-ide/use-airflow-operators.md:10]: See other title suggestion, I think this is a bit too close to the title of the doc itself

File: astro/cloud-ide/custom-cell-reference.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
+---
+sidebar_label: Custom cell configuration reference
+title: Custom cell configuration reference
+id: custom-cell-reference
+description: Learn how to configure a custom cell type as JSON.
```
Comment (jwitz) [inline astro/cloud-ide/custom-cell-reference.md:5]:
```suggestion
description: Learn how to import custom operators by configuring a custom cell type with JSON.
```

File: astro/cloud-ide/use-airflow-operators.md (hunk 2 above)
Comment (jwitz) [reply astro/cloud-ide/use-airflow-operators.md:44]: That's step 8 at the end, hopefully the end user can just use the standard operator cell steps to find the custom operator

**Pull request merged**